ci/test.yml

```diff
@@ -0,0 +1,44 @@
+image: $CI_DEPENDENCY_PROXY_DIRECT_GROUP_IMAGE_PREFIX/nixos/nix:2.3.12
+
+test:
+  stage: test
+  needs: []
+  variables:
+    SDL_VIDEODRIVER: "dummy"
+  script:
+    - nix-shell --run "./src/binaries/firmware/bin/download.sh && ./src/binaries/trezord-go/bin/download.sh"
+    - nix-shell --run "poetry install"
+    - nix-shell --run "poetry run make test"
+  after_script:
+    - cp /builds/satoshilabs/trezor/trezor-user-env/logs/debugging.log trezor-user-env-debugging.log
+    - cp /builds/satoshilabs/trezor/trezor-user-env/logs/emulator_bridge.log tenv-emulator-bridge-debugging.log
+  artifacts:
+    paths:
+      - trezor-user-env-debugging.log
+      - tenv-emulator-bridge-debugging.log
+    expire_in: 1 week
+    when: always
+
+test image:
```

Comment:
I think the right way to test the image is by running the tests directly inside. That means use the `image` directive and set the docker image to the previously built one.